### PR TITLE
Describe the invalid precomputed address issue in the tools readme

### LINKED ISSUE #193: Precompute contract address on ethers deployment
### Description

On contract deployment, most of the tools pre-compute the contract address on the client-side, based on `sender address` and `nonce`. In the Hedera ecosystem, it's not like that, where it's just the next available id. 

[https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses](
https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses)
```
ethers.utils.getContractAddress( transaction )
from + nonce
 ```
In the end, the contract deployment is not 100% compatible with the Ethereum workflow.

### Steps to reproduce

Deploy a contract with an ECDSA account and alias.

### Additional context

_No response_

### Hedera network

other

### Version

v0.1.0

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #193

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools/README.md [hunk 1]
```diff
@@ -11,3 +11,34 @@
 
 Note: On contract deployment, most of the tools (e.g. [ethers](https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses)) pre-compute the contract address on the client-side, based
 on sender address and nonce. In the Hedera ecosystem, it's not like that, where it's just the next available id.
+
+#### Option 1
```
Comment (Nana-EC) [inline tools/README.md:15]: Please update the X mark in the table to a yellow caution/exclamation.

File: tools/README.md [hunk 2]
```diff
@@ -11,3 +11,34 @@
 
 Note: On contract deployment, most of the tools (e.g. [ethers](https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses)) pre-compute the contract address on the client-side, based
 on sender address and nonce. In the Hedera ecosystem, it's not like that, where it's just the next available id.
+
```
Comment (Nana-EC) [inline tools/README.md:14]: Update the Note:" to add
"[ethers js](https://docs.ethers.io/v5/api/utils/address/#utils--contract-addresses) and therefore Hardhat usage are impacted by this address calculation difference with the details captured [here](https://github.com/ethers-io/ethers.js/discussions/3141).
An extra step to retrieve the valid Hedera contract address is required to workaround this challenge, example workarounds are provided below"

File: tools/README.md [hunk 3]
```diff
@@ -11,3 +11,34 @@
...
+#### Option 1
+```typescript
+// init the contract factory
+const factory = new ethers.ContractFactory(contractJson.abi, contractJson.bytecode, wallet);
+// deploy the contract
+let contract = await factory.deploy();
+
+// wait till the transaction is mined and get the contract address from the receipt
```
Comment (Nana-EC) [inline tools/README.md:22]:
```suggestion
// wait till the transaction has reached consensus and get the contract address from the receipt
```

File: tools/README.md [hunk 4]
```diff
@@ -11,3 +11,34 @@
...
+```
+
+#### Option 2
+```typescript
+// init the contract factory
+const factory = new ethers.ContractFactory(contractJson.abi, contractJson.bytecode, wallet);
+// deploy the contract
+let contract = await factory.deploy();
+
+// wait for transaction to be mined
```
Comment (Nana-EC) [inline tools/README.md:36]:
```suggestion
// wait for transaction to reach consensus 
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Some documentation suggestions

File: tools/README.md (hunk 2 above)
Comment (Nana-EC) [reply tools/README.md:17]: You didn't add the link to the discussion item on the etherejs repo for the "here"

**Review by Nana-EC — DISMISSED** [review]
You forget to add the link to the discussion item.
Rest looks good. Signing off so I'm not blocking once you make that change since the rest is good

**Review by Ivo-Yankov — APPROVED** [review]

**Pull request merged**